packages/server/graphql/intranetSchema/mutations/enableSAMLForDomain.ts

```diff
@@ -19,14 +19,31 @@ const enableSAMLForDomain = {
     const r = await getRethink()
     const normalizedDomain = domain.toLowerCase()
     const normalizedUrl = url.toLowerCase()
-    await r
+
+    const samlDomainExists = (await r
       .table('SAML')
-      .insert({
-        domain: normalizedDomain,
-        url: normalizedUrl,
-        metadata: metadata
-      })
-      .run()
+      .getAll(normalizedDomain, {index: 'domain'})
+      .count()
+      .run()) as number
+
+    if (samlDomainExists) {
+      await r
+        .table('SAML')
+        .update({
```

Comment:
careful, this updates every single row on the table!
you want to first `.get` the row you want to update